sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2353,7 +2353,7 @@ class Analyzer(override val catalogManager: CatalogManager)
       case q: UnaryNode if q.childrenResolved =>
         resolveSubQueries(q, q.children)
       case j: Join if j.childrenResolved =>
```

Comment:
seems we should also check `j.duplicateResolved`

Reply:
Done, but I am confused witch case will cause error.